### PR TITLE
[RSS-44] GET /announcements Endpoint

### PR DESCRIPTION
**Pull request by buffersquid**
Tested pretty thoroughly and works as expected. The only (possible) issue is that my sorting method takes place within the code and not via AWS lambda sorting, which may be inefficient in the long run once more and more announcement objects get put into the table. I wanted to sort via the database but couldn't because I 1) don't have access to lambda functions in AWS (which I think are needed to sort a table?), and 2) couldn't figure out a way to do it without messing up the tables.

### CODE REVIEW

File: amplify/backend/function/risestandstrongcb34053a/src/routes/announcements.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 const express = require('express');
+// const { getAnnouncement } = require('../../../../../../src/api');
```
Comment (jackfales) [inline amplify/backend/function/risestandstrongcb34053a/src/routes/announcements.js]:
```suggestion
```

File: amplify/backend/function/risestandstrongcb34053a/src/routes/announcements.js [hunk 2]
```diff
@@ -49,23 +51,30 @@ const router = express.Router();
...
  *                        $ref: '#/components/schemas/Announcement'
  */
 router.get('/', async (req, res) => {
-    res.end();
+    try{
+        const data = await getAnnouncements();
+        res.json(data);
+    }
+    catch (err){
+        // console.log("ERRORRfdsjklfjdskfj")
```
Comment (jackfales) [inline amplify/backend/function/risestandstrongcb34053a/src/routes/announcements.js]:
```suggestion
```

File: src/api.js [hunk 3]
```diff
@@ -6,7 +6,21 @@ const BASE_URL = 'https://1nazz9rwh9.execute-api.us-west-2.amazonaws.com/prod';
...
-}
\ No newline at end of file
+}
+
+async function getAnnouncements(userSub) {
+    await axios.get(`${BASE_URL}/${ANNOUNCEMENTS}`)
+      .catch(function (error) {
+        // handle error
+        console.log("ERROR " + error);
+      });
```
Comment (jackfales) [inline src/api.js]: Remove this, we already have error handling in the backend

File: amplify/backend/function/risestandstrongcb34053a/src/utils/aws-utils.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
+const { CodeStarNotifications } = require('aws-sdk');
```
Comment (jackfales) [inline amplify/backend/function/risestandstrongcb34053a/src/utils/aws-utils.js]: Are we using this? If not remove

File: amplify/backend/function/risestandstrongcb34053a/src/utils/aws-utils.js [hunk 5]
```diff
@@ -25,6 +28,33 @@ async function postAnnouncement(announcementBody) {
...
 
+
+async function getAnnouncements(){
+    function sortTable(data){
+        const topfive = [];
+        data.Items.forEach(element => {
+            topfive.push(element);
+        });
+        topfive.sort((a, b) => b.createdAt - a.createdAt);
+        // topfive = topfive.slice(0, 5);
```
Comment (jackfales) [inline amplify/backend/function/risestandstrongcb34053a/src/utils/aws-utils.js]: Remove leftover comments

**Review by jackfales — COMMENTED** [review]
Couple of comments to remove some code but for the most part it looks good! Request my review again when changes are made and merge conflicts are resolved.

**Comment by buffersquid** [comment]
Everything should be fixed except the merge conflicts. When I fixed them it wanted me to merge the commit but I want to get your review first.

**Review by jackfales — APPROVED** [review]
Looks good

**Pull request merged**